renderer/components/Onboarding/Steps/WalletRecover.js

```diff
@@ -20,18 +28,67 @@ class WalletRecover extends React.Component {
     this.formApi.submitForm()
   }
 
-  handleSubmit = () => {
-    const { recoverOldWallet } = this.props
+  componentDidUpdate(prevProps) {
+    const { isRecoveringWallet, walletRecoveryError } = this.props
+
+    // Handle success case.
+    if (!walletRecoveryError && !isRecoveringWallet && prevProps.isRecoveringWallet) {
+      this.handleSuccess()
+    }
+
+    // Handle failure case.
+    if (walletRecoveryError && !isRecoveringWallet && prevProps.isRecoveringWallet) {
+      this.handleError()
+    }
```

Comment:
Would be cleaner to code it like this imo:
```

    if( !isRecoveringWallet && prevProps.isRecoveringWallet){
        if(walletRecoveryError){
            this.handleError()
        }
        else {
            this.handleSuccess()
        }
    }
```

Reply:
This has been refactored